src/Simplex/Messaging/Agent/Client.hs

```diff
@@ -1905,6 +1907,97 @@ withNextSrv c userId usedSrvs initUsed action = do
     writeTVar usedSrvs $! used'
   action srvAuth
 
+data AgentServersSummary = AgentServersSummary
+  { usersServersSummary :: Map UserId ServersSummary -- UserId to be mapped to User in chat core
+    -- compute in backend to not repeat logic in UI?
+    -- requires separate type / changing summary type to accomodate many servers (e.g. lists of workers, etc.)
+    -- totalServersSummary :: ServersSummary
+  }
+  deriving (Show)
+
+data ServersSummary = ServersSummary
+  { smpServersSummary :: [SMPServerSummary],
+    xftpServersSummary :: [XFTPServerSummary]
+  }
+  deriving (Show)
+
+data SMPServerSummary = SMPServerSummary
+  { smpServer :: SMPServer,
+    usedForNewConnections :: Bool,
+    subscriptionsSummary :: Maybe SMPServerSubsSummary,
+    workersSummary :: SMPServerWorkersSummary,
+    commandsStats :: [CommandStat],
+    rcvMsgCounts :: [SMPServerRcvMsgCounts],
```

Comment:
if we want to count by connection ID, maybe map?
Why do we need to count by connection ID?

Reply:
I was trying to keep granularity of existing info and tie it to servers at the same time. New plan is to show only limited info for servers and dump the rest debug info into share